cockatrice/src/update_checker.cpp

```diff
@@ -37,18 +34,33 @@ void UpdateChecker::check()
 #if defined(Q_OS_OSX)
 bool UpdateChecker::downloadMatchesCurrentOS(QVariant build)
 {
-   return build
+    return build
             .toMap()["name"]
             .toString()
-            .contains("osx");
+            .endsWith(".dmg");
 }
+
 #elif defined(Q_OS_WIN)
 bool UpdateChecker::downloadMatchesCurrentOS(QVariant build)
 {
-       return build
-            .toMap()["name"]
-            .toString()
-            .contains("exe");
+    QString wordSize = QSysInfo::buildAbi().split('-')[2];
+    QString arch;
+    if (wordSize == "llp64") {
+        arch = "win64";
+    } else if (wordSize == "ilp32") {
+        arch = "win32";
+    } else {
+        qWarning() << "Error checking for upgrade version: wordSize is" << wordSize;
+        return false;
+    }
+
+    auto fileName = build
+        .toMap()["name"]
+        .toString();
+    // Checking for .zip is a workaround for the May 6th 2016 release
+    auto zipName = arch + ".zip";
```

Comment:
I considered that but I thought this was relatively harmless to add, and it makes it work with the release that has .zip uploaded. Remember that people will be upgrading *to* this release by using the old version so we will have to .zip this release as well - if I didn't check for zip, this release wouldn't be able to detect itself for 32bit.